src/cljs/main/quil_site/examples/floating_graph.cljs

```diff
@@ -234,33 +202,26 @@
         state (if (= (:frames-to-wait search-state) 1)
                 (x-first-search (assoc-in state [:search-state :frames-to-wait] frames-between-visits))
                 (update-in state [:search-state :frames-to-wait] dec))
-        collection (get-in state [:search-state :collection]) 
-        state (if (empty? collection) 
+        collection (get-in state [:search-state :collection])
+        state (if (empty? collection)
                 ;; when x-first-search returns empty collection, search finished, so start fresh with the other search alg
-                (infer-new-search-state (if (= :bfs (get-in state [:search-state :search-type])) :dfs :bfs) (:nodes state))
+                (merge state
+                       (infer-new-search-state (if (= :bfs (get-in state [:search-state :search-type])) :dfs :bfs) (:nodes state)))
                 state)]
     state))
 
 (defn update-state
   "quil update-state step"
   [state]
-  (update-colors 
-    (update-movement state)))
+  (-> state update-movement update-colors))
 
-(defn draw-state 
-  "quil draw-state step"
-  [state]
-  (q/background 0 0 0 0)
-  (q/no-stroke)
-  (let [nodes (:nodes state)
-        edges (extract-edges (vals nodes))
-        visiting (peek (get-in state [:search-state :collection]))] 
+(defn draw-edges [state]
+  (let [{:keys [nodes colors]} state
```

Comment:
-  (let [{:keys [nodes colors]} state
-        edges (extract-edges (vals nodes))]
+  (let [{:keys [nodes colors edges]} state]

(uses edges defined in state instead, combined with annotation above.)